packages/frontend-api/tests/Unit/Model/Token/TokenFacadeTest.php

```diff
@@ -0,0 +1,131 @@
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserFacade;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\ExpiredTokenUserMessageException;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\InvalidTokenUserMessageException;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\NotVerifiedTokenUserMessageException;
+use Shopsys\FrontendApiBundle\Model\Token\JwtConfigurationFactory;
+use Shopsys\FrontendApiBundle\Model\Token\TokenFacade;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+
+class TokenFacadeTest extends TestCase
+{
+    /**
+     * @dataProvider tokensDataProvider
+     * @param \Lcobucci\JWT\UnencryptedToken $token
+     * @param class-string|null $exceptionClass
+     */
+    public function testTokenValidation(UnencryptedToken $token, ?string $exceptionClass): void
+    {
+        $tokenFacade = $this->createTokenFacade();
+
+        if ($exceptionClass !== null) {
+            $this->expectException($exceptionClass);
+        }
+        $tokenFacade->validateToken($token);
+    }
+
+    /**
+     * @return iterable
+     */
+    public function tokensDataProvider(): iterable
+    {
+        $builderTemplate = (new Builder(new JoseEncoder(), ChainedFormatter::default()))
+            ->issuedBy('http://webserver:8080')
+            ->permittedFor('http://webserver:8080')
+            ->issuedAt(new DateTimeImmutable())
+            ->canOnlyBeUsedAfter(new DateTimeImmutable('- 10 minutes'))
+            ->expiresAt(new DateTimeImmutable('+ 10 minutes'));
+
+        $jwtConfiguration = $this->createJwtConfiguration();
+        $signer = $jwtConfiguration->signer();
+        $privateKey = $jwtConfiguration->signingKey();
+
+        $builder = clone $builderTemplate;
+        yield [
+            $builder->getToken($signer, $privateKey),
+            null,
+        ];
+
+        $builder = clone $builderTemplate;
+        yield [
+            $builder
+                ->issuedBy('http://another-server:8080')
+                ->getToken($signer, $privateKey),
+            InvalidTokenUserMessageException::class,
+        ];
+
+        $builder = clone $builderTemplate;
+        yield [
+            $builder->getToken($signer, InMemory::file(__DIR__ . '/testKeys/invalid-private.key')),
+            NotVerifiedTokenUserMessageException::class,
+        ];
+
+        $builder = clone $builderTemplate;
+        yield [
+            $builder
+                ->expiresAt(new DateTimeImmutable('- 5 minutes'))
+                ->getToken($signer, $privateKey),
+            ExpiredTokenUserMessageException::class,
+        ];
+    }
+
+    /**
+     * @return \Shopsys\FrontendApiBundle\Model\Token\TokenFacade
+     */
+    private function createTokenFacade(): TokenFacade
+    {
+        $domain = $this->createDomain();
+
+        $customerUserFacade = $this->createMock(CustomerUserFacade::class);
+        $customerUserFacade->method('addRefreshTokenChain');
```

Comment:
is this necessary when you do not set any expected outcome from the method?

Reply:
Oh, you're right. It is not. 
Fixed in 573338ddd